metricbeat/module/prometheus/remote_write/data.go

```diff
@@ -0,0 +1,98 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package remote_write
+
+// TODO this file is entirely copied from collector/data.go, refactor this
+
+import (
+	"math"
+	"time"
+
+	"github.com/prometheus/common/model"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+// PromEvent stores a set of one or more metrics with the same labels
+type PromEvent struct {
+	data      common.MapStr
+	labels    common.MapStr
+	timestamp time.Time
+}
+
+// LabelsHash returns a repeatable string that is unique for the set of labels in this event
+func (p *PromEvent) LabelsHash() string {
+	return p.labels.String()
+}
+
+func samplesToEvents(metrics model.Samples) map[string]mb.Event {
```

Comment:
I understand Samples will always be untyped right? For example, we won't be able to differenciate a histogram metric from a gauge

Reply:
Prometheus sends data with [writerequest](https://godoc.org/github.com/prometheus/prometheus/prompb#WriteRequest) using this [proto](https://github.com/prometheus/prometheus/blob/master/prompb/remote.proto). Not sure if we can retrieve the type of the metric in an easy way 🤔

Reply:
I guess this is going to be an issue with histograms? Should we do a best-effort guess if it includes metrics with some special labels and/or suffixes?